index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
I'm seeing what I can do about supporting the path logic and wondering why you decided to do this? I would want the prefix to be included with all because when the group is optional the prefix shouldn't be output.